Wikipedia/Code/WMFAppViewController.m

```diff
@@ -182,6 +199,8 @@ - (void)resumeApp {
             [[WMFAlertManager sharedInstance] showAlert:@"You have been notified" sticky:NO dismissPreviousAlerts:NO tapCallBack:NULL];
         }
     }
+
+    [self handleSelectionOfShortcutItemIfNecessary];
```

Comment:
Doesn't this line need to be before line 177: 

`if ([self shouldShowHomeScreenOnLaunch]) {`

I don't think we `shouldShowHomeScreenOnLaunch` or `shouldShowLastReadArticleOnLaunch` if we are going to respect the app shortcut.

So seems like we should start this if/else chain with:

`if(self.shortcutItemSelectedAtLaunch){`